### PR TITLE
[WIP] Circleci project setup

### PR DESCRIPTION
**Pull request by darshshah**
- Update the circleci config file to add python tests.
- Added pytest-html to save the test reports from ci

### CODE REVIEW

**Comment by darshshah** [comment]
Python tests are timing out after 10mins. I'll add a minimum threshold so that we don't hit this again.

**Comment by jakemannix** [comment]
> Python tests are timing out after 10mins. I'll add a minimum threshold so that we don't hit this again.

hey @lastmansleeping, how long should the python tests take?

**Comment by lastmansleeping** [comment]
> > Python tests are timing out after 10mins. I'll add a minimum threshold so that we don't hit this again.
> 
> hey @lastmansleeping, how long should the python tests take?

On my laptop `python3 -m pytest` takes about 3 minutes to complete.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine. See: https://circleci.com/docs/2.0/configuration-reference
+# Reference doc: https://circleci.com/docs/2.0/project-walkthrough/
```
Comment (balikasg) [inline .circleci/config.yml:2]: Thanks for adding this. If you thought that nobody would read it, guess what: I am reading it.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          key: python-deps-{{ checksum "python/requirements.txt" }}
+          paths:
+            - "venv"
+      - run:
+          name: Run Python tests
+          command: |
+            . venv/bin/activate
+            cd python
+            python3 -m pytest --html=pytest_report.html
+          no_output_timeout: 30m  # Time for running the tests before we timeout
```
Comment (balikasg) [inline .circleci/config.yml:56]: Are we on a paid version or on the free one? I am clicking on the "$" sign on the left navigation bar but "Something unexpected happens"

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      - store_artifacts:
+          path: python/pytest_report.html
+      - store_test_results:
+          path: python/pytest_report.html
+
+workflows:
```
Comment (balikasg) [inline .circleci/config.yml:62]: Nice! 
I did not know `circleci` but it is a nice and clean interface.

File: .circleci/config.yml (hunk 1 above)
Comment (balikasg) [reply .circleci/config.yml:2]: Well actually by following this I was able to understand everything that is happening in this config.

Comment (darshshah) [reply .circleci/config.yml:2]: Glad it helped.

File: .circleci/config.yml (hunk 2 above)
Comment (darshshah) [reply .circleci/config.yml:56]: We are a part of Salesforce's open-source organization. So it takes care of the credits/billing.

**Pull request merged**